### DISCUSSION TITLE
感觉 with autocast(enabled=hps.train.bf16_run, dtype=torch.bfloat16)用的有点混乱

### DISCUSSION THREAD

**Discussion by NiHaoUCAS** [fishaudio/Bert-VITS2#306] [General]
https://github.com/fishaudio/Bert-VITS2/blob/Refactor/train_ms.py#L459用了一个autocast，但是在L510和L532又用了两个嵌套的autocast。 如果是这样的话， L541的scaler.scale(loss_dur_disc_all).backward()就是用的bf16了。

**Comment by Stardust-minus** [comment]
基本上继承vits2原版代码，有更好的用法欢迎pr
